code/__HELPERS/matrices.dm

```diff
@@ -63,3 +63,92 @@
 //The Y pixel offset of this matrix
 /matrix/proc/get_y_shift()
 	. = f
+
+/datum/ColorMatrix
+	var/list/matrix
+	var/combined = 1
+	var/const/lumR = 0.3086 //  or  0.2125
+	var/const/lumG = 0.6094 //  or  0.7154
+	var/const/lumB = 0.0820 //  or  0.0721
+
+/datum/ColorMatrix/New(mat, contrast = 1, brightness = null)
+	..()
+	if(istext(mat))
+		SetColor(mat, contrast, brightness)
+	else if(isnum(mat))
+		SetSaturation(mat, contrast, brightness)
+	else
+		matrix = mat
+
+/datum/ColorMatrix/proc/Reset()
+	matrix=list(1,0,0,
+				0,1,0,
+				0,0,1)
+
+/datum/ColorMatrix/proc/Get(contrast = 1)
+	var/list/mat = matrix
+	mat = mat.Copy()
+
+	for(var/i = 1 to min(mat.len, 12))
+		mat[i] *= contrast
+	return mat
+
+/datum/ColorMatrix/proc/SetSaturation(s, c = 1, b = null)
```

Comment:
Use better variable names here, not just single-letter ones. I have no idea what these terms mean at a glance.